tools/jenkins/run_distribution.sh

```diff
@@ -75,58 +75,48 @@ elif [ "$platform" == "macos" ]; then
     mkdir -p $brew_root
     git clone https://github.com/Homebrew/homebrew.git $brew_root
 
-    # Install grpc via homebrew
-    #
     # The temp $PATH env variable makes sure we are operating at the right copy of
     # temp homebrew installation, and do not interfere with the system's main brew
     # installation.
-    #
-    # TODO: replace the next section with the actual homebrew installation script
-    # i.e. curl -fsSL https://goo.gl/getgrpc | bash -s $language
-    # need to resolve a bunch of environment and privilege issue on the jenkins
-    # mac machine itself
-    export OLD_PATH=$PATH
+    export ORIGINAL_PATH=$PATH
     export PATH=$brew_root/bin:$PATH
-    cd $brew_root
-    brew tap homebrew/dupes
-    brew install zlib
-    brew install openssl
-    brew tap grpc/grpc
-    brew install --without-python google-protobuf
-    brew install grpc
-    brew list -l
 
-    # Install per-language modules/extensions on top of core grpc
-    #
-    # If a command below needs root access, the binary had been added to
-    # /etc/sudoers. This step needs to be repeated if we add more mac instances
-    # to our jenkins project.
-    #
-    # Examples (lines that needed to be added to /etc/sudoers):
-    # + Defaults        env_keep += "CFLAGS CXXFLAGS LDFLAGS enable_grpc"
-    # + jenkinsnode1 ALL=(ALL) NOPASSWD: /usr/bin/pecl, /usr/local/bin/pip,
-    # +   /usr/local/bin/npm
+    # Set up right environment for each language
+    case $language in
+      *python*)
+        virtualenv venv
```

Comment:
if last  run of the distribution test failed, will we find dirty environment at this point? perhaps we could put the cleanup as a first step before we actually do anything. In that case, if it's impossible to cleanup and start with a fresh environment, the test will fail early.